astropy/units/core.py

```diff
@@ -1409,6 +1406,46 @@ def __repr__(self):
                          [']'])
                 return '\n'.join(lines)
 
+        def _repr_html_(self):
+            """
+            Outputs a HTML table representation within Jupyter notebooks.
+            """
+            if len(self) == 0:
+                return "<p>{}</p>".format(self.NO_EQUIV_UNITS_MSG)
+            else:
+                # HTML tags to use to compose the table in HTML
+                blank_table = '<table style="width:100%">{}</table>'
+                blank_row_container = "<tr>{}</tr>"
+                heading_row_content = "<th>{}</th>" * self.ROW_LEN
+                data_row_content = "<td>{}</td>" * self.ROW_LEN
```

Comment:
Since ``Table`` objects look nice in Jupyter notebooks, I wonder if it'd be worth trying to just construct a table here and then using that for ``_repr_html_``? (to avoid duplicating the construction of the HTML table). In fact you could use that for both ``__repr__`` and ``_repr_html_``? I wonder if there is a reason this wasn't done before though, so I'll ping @mhvk, @adrn, and @taldcroft in case they know.

Reply:
:+1: on using `Table`!  One detail is that the default HTML repr has info about the table class and dtypes, and worries about the max number of lines to display.  So you can fix that up with a subclass like so:
```
class MinimalReprTable(Table):
    def _base_repr_(self, *args, **kwargs):
        kwargs['descr_vals'] = []
        kwargs['max_lines'] = -1
        kwargs['show_dtype'] = False
        return super()._base_repr_(*args, **kwargs)

    def __repr__(self):
        return '\n'.join(self.pformat(max_width=-1, max_lines=-1))
```

Reply:
@taldcroft - ah this raises an interesting point, currently it looks like ``EquivalentUnitList`` inherits from ``List``, and we can't change that inheritance without breaking user code. So I wonder if returning a Table repr will actually confuse things for users who will then expect it to behave as a table. That would argue for keeping it as it is in this PR - looking much better esthetically but different from Table?

Reply:
I would argue that consistency in the style (CSS)  of outputted tabular information across astropy is definitely a plus, not a minus.  Agreed that having an class where the repr implies a different object type is not a great thing, but I suspect that only experienced users would recognize the style difference in this PR vs. standard Table style and immediately guess that the `EquivalentUnitList` is not a Table.

One comment I would make on this PR is that I don't that 100% width is the right thing if you happen to be working on 27" screen with a full-width browser open.  With that change then the difference between this output and `MinimalReprTable` is fairly subtle.

Reply:
@taldcroft - ok I see your point, and I agree then that maybe it does indeed make sense to use the Table functionality to build the repr and html repr. We just need to be careful to not include any mention of 'Table' in the output.

@sadielbartholomew - would you be able to try and se if you can leverage the Table repr functionality to avoid duplicating HTML formatting code? No problem if you don't have time of course, but just wanted to check.

Reply:
Maybe try not to import Table at the module level? I don't want the circular import issue to creep in later, like what I see going on with Time.

Reply:
Yes, good point. In `units` we should import nothing else of astropy at the module level - it is needed for almost everything else.

Reply:
Thanks for the input all, & my apologies, I am completely new to the project & did not realise there was already some code for HTML tables I could have used (or though to check for it).

Saying that, I'm getting mixed messages, so please can someone outline is there is a consensus on whether or not to change it to use a ``Table``?

> would you be able to try and se if you can leverage the Table repr functionality to avoid duplicating HTML formatting code? No problem if you don't have time of course, but just wanted to check.

Admittedly I am short on time, so while I am happy to change to an approach using code from ``Table`` if there's consensus for that, I would need to wait until (very) early November to make a large-scale change realistically (I want to complete Hacktober & started late so have three other PRs to finalise). However I'm happy to deal with any conflict resolution that might result in.

I appreciate I am an outsider to the project & trust all your judgements, but from reading your comments it does seem to me like instead using ``Table`` would take as many lines of code, be less clear/comprehensible, & also require an awkward import, & only defining HTML tags is really duplicated, so I don't personally see any reason to change from what I have. (Happy to go with whatever you decide, though!)

Reply:
@EdwardBetts - thanks for your patience. I would be fine with merging as is - it is a nice improvement and we can do the `Table` version later if need be.

Reply:
Agreed.  Perfect is the enemy of good!